src/gui/wizard/xyzcalib.c

```diff
@@ -80,9 +79,8 @@ int xyzcalib_xy_measure(int16_t id_body, xyzcalib_screen_t *p_screen, xyzcalib_d
             "points");
         window_set_icon_id(p_screen->icon.win.id, IDR_PNG_wizard_icon_measure);
         marlin_gcode("G29");
-        while (!(marlin_busy() || marlin_motion()))
-            osDelay(10);
-    } else if (!(marlin_busy() || marlin_motion()))
+        marlin_event_clr(MARLIN_EVT_CommandEnd);
+    } else if (marlin_event_clr(MARLIN_EVT_CommandEnd))
```

Comment:
error-prone and hard-to-read - I'm missing the { } for this if-block.

Reply:
{} is here, maybe the diff is confusing